sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Blob/ContainerRegistryBlobClient.cs

```diff
@@ -621,15 +598,80 @@ public virtual async Task<Response<DownloadBlobResult>> DownloadBlobAsync(string
             scope.Start();
             try
             {
-                ResponseWithHeaders<Stream, ContainerRegistryBlobGetBlobHeaders> blobResult = await _blobRestClient.GetBlobAsync(_repositoryName, digest, cancellationToken).ConfigureAwait(false);
+                return await DownloadBlobInternalAsync(digest, true, cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
 
-                // TODO: Compute digest asynchronously, if large.
-                if (!ValidateDigest(blobResult.Value, digest))
-                {
-                    throw new RequestFailedException("The requested digest does not match the digest of the received manifest.");
-                }
+        private async Task<Response<DownloadBlobResult>> DownloadBlobInternalAsync(string digest, bool async, CancellationToken cancellationToken)
+        {
+            ResponseWithHeaders<Stream, ContainerRegistryBlobGetBlobHeaders> blobResult = async ?
+                await _blobRestClient.GetBlobAsync(_repositoryName, digest, cancellationToken).ConfigureAwait(false) :
+                _blobRestClient.GetBlob(_repositoryName, digest, cancellationToken);
+
+            var contentDigest = BlobHelper.ComputeDigest(blobResult.Value);
+            ValidateDigest(contentDigest, digest);
+
+            BinaryData data = async ?
+                await BinaryData.FromStreamAsync(blobResult.Value, cancellationToken).ConfigureAwait(false) :
+                BinaryData.FromStream(blobResult.Value);
+
+            return Response.FromValue(new DownloadBlobResult(digest, data), blobResult.GetRawResponse());
+        }
+
+        /// <summary>
+        /// Download a blob to a passed-in destination stream.
+        /// </summary>
+        /// <param name="digest">The digest of the blob to download.</param>
+        /// <param name="destination">Destination for the downloaded blob.</param>
+        /// <param name="options">Options for the blob download.</param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <returns>The raw response corresponding to the final GET blob chunk request.</returns>
+        public virtual Response DownloadBlobTo(string digest, Stream destination, DownloadBlobToOptions options = default, CancellationToken cancellationToken = default)
```

Comment:
something to consider: blob size and digest come from OciBlobDescriptor, it'd nice to do something like:
```
var manifest = client.DownloadManifest("foo").GetManifest();
foreach (var layer : manifest.Layers) {
 client.DownloadBlobTo(layer, stream, cancellationToken) ;
}
```

then `DownloadBlobTo` options are only needed for chunk size. But I'm not sure if chunk size makes sense on per-request basis.


So alternative proposal to consider is  
- have 2 overloads for `DownloadBlobInternalAsync`: 
  - `(string digest, Stream dest, CancellationToken cancellationToken)`
  - `(OciBlobDescriptor blobDescriptor, Stream dest,  CancellationToken cancellationToken)`
-  add chunk size as client-level property (or even hardcode one for now)
- If in future we need download options, we can always add them

Reply:
if we end up not needing blobSize ahead of time (see my other comment) I still would like to consider chunk size on the client level (or hardcoded)

Reply:
Addressed.